q2_types_genomics/genome_data/_format.py

```diff
@@ -0,0 +1,158 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2021, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import qiime2.plugin.model as model
+from q2_types.feature_data import DNAFASTAFormat, ProteinFASTAFormat
+from qiime2.core.exceptions import ValidationError
+
+from ..plugin_setup import plugin
+
+GenesDirectoryFormat = model.SingleFileDirectoryFormat(
+    'GenesDirectoryFormat',
+    r'genes[0-9]+\.(fa|fasta)$',
```

Comment:
SingleFileDirectoryFormat is the wrong element to use here, as it presumes exactly one file in a directory (it's really just a shortcut, since the single file can be easily predicted).

In this case, you will want to use a full DirectoryFormat with FileCollection.

Reply:
It _did_ feel like cheating... Thanks for pointing that out - updated according to your suggestion.